### ISSUE TITLE
DataContext Ctor question

### ISSUE THREAD

**Issue by wanaxe** [linq2db/linq2db#3591]
### Issues, and Steps to reproduce

Just upgraded from v3.6.0 to v4.0.1. As there're some breaking changes in v4.0.x, I'm not sure if this is a bug or I did something wrong.

For v3.6.0, my code worked well, and was like

```c#
var dataProvider = new SqlServerDataProvider(ProviderName.SqlServer, 
	SqlServerVersion.v2012, SqlServerProvider.MicrosoftDataSqlClient);
var dataContext = new DataContext(dataProvider, GetCurrentConnectionString());
```
For v4.0.1, the code is changed to like

```c#
var dataProvider = SqlServerTools.GetDataProvider(SqlServerVersion.v2012, 
	SqlServerProvider.MicrosoftDataSqlClient);
var dataContext = new DataContext(dataProvider, GetCurrentConnectionString())
```

The `DataContext` Ctor throws a `ArgumentNullException`, for paramater 'providerName'. But the code calls  `DataContext(IDataProvider dataProvider, string connectionString)`, not `DataContext( string providerName, string connectionString)`.

Insteadly, if using the later Ctor, it works.

```c#
var dataProvider = SqlServerTools.GetDataProvider(SqlServerVersion.v2012, 
	SqlServerProvider.MicrosoftDataSqlClient);
var dataContext = new DataContext(dataProvider.Name, GetCurrentConnectionString())
```
### Exception message:
Stack trace:
```
System.ArgumentNullException
  HResult=0x80004003
  Message=Value cannot be null. (Parameter 'providerName')
  Source=linq2db
  StackTrace:
   at LinqToDB.Configuration.LinqToDBConnectionOptionsBuilder.UseConnectionString(String providerName, String connectionString)
   at LinqToDB.DataContext..ctor(LinqToDBConnectionOptions options)
   at LinqToDB.DataContext..ctor(IDataProvider dataProvider, String connectionString)
```

### Environment details
linq2db version: 4.0.1
Database Server: SQL Server 2014
Operating system: Windows 11 Pro
.NET Framework: .NET 6

**Comment by viceroypenguin** [comment]
Yes, there was a bug; should be fixed in 4.0.2. Thanks for reporting!

**Comment by wanaxe** [comment]
Great. Than you so much!
